as-nobody.c

```diff
@@ -0,0 +1,30 @@
+#include <pwd.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <grp.h>
+
+int
+main(int argc, char *argv[])
+{
+  struct passwd *pw = getpwnam("nobody");
+  if (pw == NULL) {
+    perror("getpwnam");
+    return EXIT_FAILURE;
+  }
+  if (setgroups(0, NULL) < 0) {
+    perror("setgroups");
+    return EXIT_FAILURE;
+  }
+  if (setregid(pw->pw_gid, pw->pw_gid) < 0) {
+    perror("setgroups");
+    return EXIT_FAILURE;
+  }
+  if (setreuid(pw->pw_uid, pw->pw_uid) < 0) {
+    perror("setreuid");
+    return EXIT_FAILURE;
+  }
+  execvp(argv[1], argv);
```

Comment:
Shouldn't this be?
```cpp
  execvp(argv[1], argv + 1);
```

Reply:
Negative. Per the manpage:

The execv(), execvp(), and execvpe() functions provide an array of pointers to null-terminated strings that represent the argument list available to the new program. The first argument, by convention, should point to the filename associated with the file being executed.

Reply:
Ayyy, good call. (sorry if you saw deleted, reread code and suggestion and thought better of it) Fixed up now with quick edit.